src/test/app/claims/models/claim.ts

```diff
@@ -148,4 +154,73 @@ describe('Claim', () => {
       expect(claim.defendantOffer).to.be.eq(undefined)
     })
   })
+
+  describe('claimStatus', () => {
+
+    it('should return ELIGIBLE_FOR_CCJ if defendant has not responded before the deadline', () => {
+      const claim = buildClaim()
+      claim.responseDeadline = moment().subtract(2, 'days')
+
+      expect(claim.status).to.be.equal(ClaimStatus.ELIGIBLE_FOR_CCJ)
+    })
+
+    it('should return CCJ_REQUESTED when a claimant requests a CCJ', () => {
+      const claim = buildClaim()
+      claim.countyCourtJudgmentRequestedAt = moment()
+
+      expect(claim.status).to.be.equal(ClaimStatus.CCJ_REQUESTED)
+    })
+
+    it('should return OFFER_SUBMITTED if an offer has been submitted', () => {
+      const claim = buildClaim()
+      claim.settlement = new Settlement()
+      claim.response = {
+        responseType: ResponseType.FULL_DEFENCE,
+        defenceType: DefenceType.DISPUTE,
+        defence: 'defence reasoning',
+        freeMediation: FreeMediationOption.YES,
+        defendant: new Individual().deserialize(individual)
+      }
+
+      expect(claim.status).to.be.equal(ClaimStatus.OFFER_SUBMITTED)
+    })
+
+    it('should return MORE_TIME_REQUESTED when more time is requested', () => {
+      const claim = buildClaim()
+      claim.moreTimeRequested = true
+
+      expect(claim.status).to.be.equal(ClaimStatus.MORE_TIME_REQUESTED)
+    })
+
+    it('should return FREE_MEDIATION when defendant has rejected the claim and asked for free mediation', () => {
+      const claim = buildClaim()
+      claim.response = {
+        responseType: ResponseType.FULL_DEFENCE,
+        defenceType: DefenceType.DISPUTE,
+        defence: 'defence reasoning',
+        freeMediation: FreeMediationOption.YES,
+        defendant: new Individual().deserialize(individual)
+      }
+      expect(claim.status).to.be.equal(ClaimStatus.FREE_MEDIATION)
+    })
+
+    it('should return CLAIM_REJECTED when an individual defendant has rejected the claim with no free mediation', () => {
+      const claim = buildClaim()
+      claim.response = {
+        responseType: ResponseType.FULL_DEFENCE,
+        defenceType: DefenceType.DISPUTE,
+        defence: 'defence reasoning',
+        freeMediation: FreeMediationOption.NO,
+        defendant: new Individual().deserialize(individual)
+      }
+      expect(claim.status).to.be.equal(ClaimStatus.CLAIM_REJECTED)
+    })
```

Comment:
It would be nice if we had a test case for `NO_RESPONSE` as well, just to be sure how this component behaves in all possible scenarios.

Reply:
NO_RESPONSE test added now 👍